homeassistant/components/tplink/siren.py

```diff
@@ -0,0 +1,68 @@
+"""Support for TPLink hub alarm."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from kasa import Device, Module
+
+from homeassistant.components.siren import SirenEntity, SirenEntityFeature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, async_refresh_after
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up siren entities."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    device = parent_coordinator.device
+
+    if Module.Alarm in device.modules:
+        async_add_entities([TPLinkSirenEntity(device, parent_coordinator)])
+
+
+class TPLinkSirenEntity(CoordinatedTPLinkEntity, SirenEntity):
+    """Representation of a tplink hub alarm."""
+
+    _attr_has_entity_name = True
```

Comment:
```suggestion
```
Is already set on `CoordinatedTPLinkEntity`

Reply:
Good catch, removed!